moda-contracts/src/Releases.sol

```diff
@@ -0,0 +1,204 @@
+ * @dev This allows artists or labels to create their own release tokens
+ * access is set to curated by default which means that only release admins
+ * can create releases. The contract can be set to open by the default admin
+ * which enables any user to create a release.
+ */
+
+contract Releases is
+    IReleases,
+    ERC1155SupplyUpgradeable,
+    ERC1155HolderUpgradeable,
+    AccessControlEnumerableUpgradeable,
+    ERC2981Upgradeable,
+    ReentrancyGuardUpgradeable
+{
+    // State Variables
+    address _catalog;
+    address _splitsFactory;
+
+    string public name;
+    string public symbol;
+
+    uint256 public numberOfReleases;
+
+    bool _isOpen;
+
+    mapping(uint256 => string) _uris;
+
+    /// @dev users with a Releases admin role can create releases with a curated contract
+    bytes32 public constant RELEASE_ADMIN_ROLE = keccak256("RELEASE_ADMIN_ROLE");
+
+    // Errors
+    error CannotBeZeroAddress();
+    error InvalidAmount();
+    error FieldCannotBeEmpty(string field);
+    error CallerIsNotReleaseAdmin();
+    error InvalidTokenId();
+    error CallerDoesNotHaveTrackAccess();
+
+    /**
+     * @dev Constructor
+     * @param isInitializerDisabled - Whether the initializer is disabled or not
+     */
+    constructor(bool isInitializerDisabled) {
+        if (isInitializerDisabled) {
+            _disableInitializers();
+        }
+    }
+
+    // External Functions
+
+    /**
+     * @inheritdoc IReleases
+     */
+    function initialize(
+        address admin,
+        address[] calldata releaseAdmins,
+        string calldata name_,
+        string calldata symbol_,
+        address catalog,
+        address splitsFactory
+    ) external initializer {
+        __ERC1155_init("");
+        if (admin == address(0)) revert CannotBeZeroAddress();
+        if (bytes(name_).length == 0) revert FieldCannotBeEmpty("name");
+        if (bytes(symbol_).length == 0) revert FieldCannotBeEmpty("symbol");
+        _grantRole(DEFAULT_ADMIN_ROLE, admin);
+        for (uint8 i = 0; i < releaseAdmins.length; i++) {
+            _grantRole(RELEASE_ADMIN_ROLE, releaseAdmins[i]);
+        }
+        name = name_;
+        symbol = symbol_;
+        _catalog = catalog;
+        _splitsFactory = splitsFactory;
+    }
+
+    /**
+     * @inheritdoc IReleases
+     */
+    function create(
+        address receiver,
+        uint96 royaltyAmount,
+        address beneficiary,
+        /// will be an array of addresses to pass to splits factory
+        string calldata uri_,
+        uint256 amount,
+        string[] calldata trackIds
+    ) external nonReentrant {
+        if (_isOpen) {
+            _requireCallerHasTrackAccess(trackIds);
+        } else if (!hasRole(RELEASE_ADMIN_ROLE, msg.sender)) {
+            revert CallerIsNotReleaseAdmin();
+        }
+        if (royaltyAmount > 2_000) revert InvalidAmount();
+
+        numberOfReleases++;
+        _uris[numberOfReleases] = uri_;
+        /// Create splits address here
```

Comment:
Let's create a ISplitsFactory with `create`. The initializer should take the interface, not the address. Then you can just call `_splitsFactory.create(...params)` right here. We can define what it really does later

Reply:
Done